src/middlewared/middlewared/plugins/virt/global.py

```diff
@@ -0,0 +1,337 @@
+                'name': 'eth0',
+                'network': INCUS_BRIDGE,
+                'type': 'nic',
+            }
+        else:
+            nic = {
+                'name': 'eth0',
+                'type': 'nic',
+                'nictype': 'bridged',
+                'parent': config['bridge'],
+            }
+
+        if import_storage:
+            payload = {
+                'pools': [{
+                    'config': {'source': config['dataset']},
+                    'description': '',
+                    'name': 'default',
+                    'driver': 'zfs',
+                }],
+            }
+            result = await incus_call('internal/recover/validate', 'post', {'json': payload})
+            if result.get('status') == 'Success':
+                if result['metadata']['DependencyErrors']:
+                    raise CallError('Missing depedencies: ' + ', '.join(result['metadata']['DependencyErrors']))
+                result = await incus_call('internal/recover/import', 'post', {'json': payload})
+                if result.get('status') != 'Success':
+                    raise CallError(result.get('error'))
+            else:
+                raise CallError('Invalid storage')
+
+        result = await incus_call('1.0/profiles/default', 'put', {'json': {
+            'config': {},
+            'description': 'Default TrueNAS profile',
+            'devices': {
+                'root': {
+                    'path': '/',
+                    'pool': 'default',
+                    'type': 'disk',
+                },
+                'eth0': nic,
+            },
+        }})
+        if result.get('status') != 'Success':
+            raise CallError(result.get('error'))
+
+    @private
+    @job()
+    async def reset(self, job):
+        config = await self.config()
+
+        if await self.middleware.call('service.started', 'incus'):
+            # Stop running instances
+            for instance in await self.middleware.call('virt.instances.query', [('status', '=', 'RUNNING')]):
+                await (await self.middleware.call('virt.instances.state', instance['id'], 'STOP', True)).wait()
+
+            if await self.middleware.call('virt.instances.query', [('status', '=', 'RUNNING')]):
+                raise CallError('Failed to stop instances')
+
+        if not config['bridge']:
+            # Make sure we delete in case it exists
+            try:
+                await run(['ip', 'link', 'show', INCUS_BRIDGE], check=True)
+            except subprocess.CalledProcessError:
+                pass
+            else:
+                await run(['ip', 'link', 'delete', INCUS_BRIDGE], check=True)
+
+        await self.middleware.call('service.stop', 'incus')
+        if await self.middleware.call('service.started', 'incus'):
+            raise CallError('Failed to stop virtualization service')
+
+        # Have incus start fresh
+        await run(['rm', '-rf', '--one-file-system', INCUS_PATH], check=True)
```

Comment:
Instead of fork+exec'ing here, you can do something like:
```
await self.middleware.run_in_thread(shutil.rmtree, INCUS_PATH)

Reply:
That will traverse filesystems, no? `--one-file-system`

Reply:
Added a comment in code